lldb/source/Plugins/ExpressionParser/Swift/SwiftUserExpression.cpp

```diff
@@ -285,8 +285,41 @@ bool SwiftUserExpression::Parse(DiagnosticManager &diagnostic_manager,
                                  "couldn't start parsing (no target)");
     return false;
   }
+
+  StackFrame *frame = exe_ctx.GetFramePtr();
+  if (!frame) {
+    LLDB_LOG(log, "  [SUE::Parse] Null stack frame");
+    return false;
+  }
+
+  // Make sure the target's SwiftASTContext has been setup before doing any
+  // Swift name lookups.
+  llvm::Optional<SwiftASTContextReader> maybe_swift_ast_ctx =
+      target->GetScratchSwiftASTContext(err, *frame);
+  if (!maybe_swift_ast_ctx) {
+    LLDB_LOG(log, "  [SUE::Parse] NULL Swift AST Context");
+    return false;
+  }
+
+  SwiftASTContext *swift_ast_ctx = maybe_swift_ast_ctx->get();
+
   if (auto *persistent_state = GetPersistentState(target, exe_ctx)) {
-    persistent_state->AddHandLoadedModule(ConstString("Swift"));
+
+    ConstString module_name("Swift");
```

Comment:
This code looks very similar to what was added in `SwiftExpressionParser.cpp`. Any way to factor this out and propagate the error in one case and passing it to LLDB_LOG in the other?

Reply:
Indeed, they pretty much do the same thing: Load the Swift Standard Library Module.
May be we should add an affordance to `SwiftASTContext` for that, what do think @adrian-prantl ?

Reply:
Sure, that would make sense!

Reply:
I ended up calling `SwiftASTContext::GetModule` instead ... It seemed unnecessary to add a new method for such a specific task when there is already a method that does the job.